os/arch/arm/src/armv7-m/Kconfig

```diff
@@ -5,6 +5,112 @@
 
 comment "ARMV7M Configuration Options"
 
+config ARMV7M_HAVE_ICACHE
+	bool
+	default n
+
+config ARMV7M_HAVE_DCACHE
+	bool
+	default n
+
+config ARMV7M_LAZYFPU
+	bool "Lazy FPU storage"
+	default n
+	depends on ARCH_HAVE_LAZYFPU
+	---help---
+		There are two forms of the common vector logic.  There are pros and
+		cons to each option:
+
+		1) The standard common vector logic exploits features of the ARMv7-M
+		   architecture to save the all of floating registers on entry into
+		   each interrupt and then to restore the floating registers when
+		   the interrupt returns.  The primary advantage to this approach is
+		   that floating point operations are available in interrupt
+		   handling logic.  Since the volatile registers are preserved,
+		   operations on the floating point registers by interrupt handling
+		   logic has no ill effect.  The downside is, of course, that more
+		   stack operations are required on each interrupt to save and store
+		   the floating point registers.  Because of the some special
+		   features of the ARMv-M, this is not as much overhead as you might
+		   expect, but overhead nonetheless.
+
+		2) The lazy FPU common vector logic does not save or restore
+		   floating point registers on entry and exit from the interrupt
+		   handler. Rather, the floating point registers are not restored
+		   until it is absolutely necessary to do so when a context switch
+		   occurs and the interrupt handler will be returning to a different
+		   floating point context.  Since floating point registers are not
+		   protected, floating point operations must not be performed in
+		   interrupt handling logic.  Better interrupt performance is be
+		   expected, however.
+
+		By default, the "standard" common vector logic is build.  This
+		option selects the alternate lazy FPU common vector logic.
+
+config ARMV7M_USEBASEPRI
```

Comment:
This has already been defined in os/arch/arm/Kconfig. 
But, I think this is the right file to define ARMV7M_BASEPRI.
Kindly remove ARMV7M_BASEPRI_WAR from os/arch/arm/Kconfig